awx/main/scheduler/task_manager.py

```diff
@@ -598,6 +536,8 @@ def process_tasks(self, all_sorted_tasks):
         self.process_running_tasks(running_tasks)
 
         pending_tasks = [t for t in all_sorted_tasks if t.status == 'pending']
+        dependencies = self.generate_dependencies(pending_tasks)
```

Comment:
Why not only pass in tasks that have `dependencies_processed=False`?

Otherwise, you're attempting updates to _lots_ of rows _every_ 30 seconds. Even if they are no-ops, that sounds like a recipe for database conflicts.

Reply:
good catch, fixed it.